src/StreamJsonRpc/Reflection/RpcTargetInfo.cs

```diff
@@ -497,18 +497,15 @@ void ActOn([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type
 #if !NET10_0_OR_GREATER
     [UnconditionalSuppressMessage("Trimming", "IL2065:UnrecognizedReflectionPattern", Justification = "false positive: https://github.com/dotnet/linker/issues/1731")]
 #endif
-    private static IReadOnlyList<EventInfo> GetEventInfos([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TypeInfo exposedMembersOnType)
+    private static IReadOnlyList<EventInfo> GetEventInfos([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicEvents | DynamicallyAccessedMemberTypes.NonPublicEvents | DynamicallyAccessedMemberTypes.Interfaces)] TypeInfo exposedMembersOnType)
```

Comment:
Can you try building this targeting `net10.0` and publishing the AOT compatibility test against `net10.0`? I'm concerned that this will now produce trim warnings since it is no longer annotated with `.All`.

Reply:
You're right about that. .NET 10 also flagged two more IL2060 warnings against JsonMessageFormatter around (wait for it...) the failure you already experienced at runtime #1230. So apparently there's an analyzer hole that .NET 10 fills.